autopush/web/base.py

```diff
@@ -283,12 +284,26 @@ def _router_fail_err(self, fail):
                 self.log.debug(format="Success", status_code=exc.status_code,
                                logged_status=exc.logged_status or 0,
                                client_info=self._client_info)
+                success = True
+                self.metrics.increment('notification.message.success',
+                                       tags=['destination:"Direct"',
+                                             'router:{}'.format(router_type),
+                                             'vapid:{}'.format(vapid is not
+                                                               None)]
+                                       )
             elif 400 <= exc.status_code < 500:
                 self.log.debug(format="Client error",
                                status_code=exc.status_code,
                                logged_status=exc.logged_status or 0,
                                errno=exc.errno or 0,
                                client_info=self._client_info)
+        if not success:
+            self.metrics.increment('notification.message.error',
+                                   tags=[
+                                        "code:{}".format(exc.status_code),
+                                        "router:{}".format(router_type),
+                                        "vapid:{}".format(vapid is True)
```

Comment:
there's still 2 bogus "vapid is True" in this patch: here and in webpush _router_completed. actually, both of these methods don't even need the full vapid dict. they just need a bool flag

webpush.py can be repsonsible for "vapid = jwt is not None" and pass that along to them